utils/logs.js

```diff
@@ -0,0 +1,122 @@
+const admin = require("firebase-admin");
+const { logType } = require("../constants/logs");
+const usersService = require("../services/dataAccessLayer");
+const { EXTENSION_REQUEST_STATUS } = require("../constants/extensionRequests");
+
+async function getUsersListFromLogs(allLogs) {
+  const userIds = [];
+  for (const log of allLogs) {
+    if (!userIds.includes(log.meta.userId || log.meta.createdBy)) {
+      userIds.push(log.meta.userId || log.meta.createdBy);
+    }
+  }
+  return await usersService.fetchUsersForKeyValues(admin.firestore.FieldPath.documentId(), userIds);
+}
+
+function formatLogsForFeed(logs, usersMap) {
+  switch (logs.type) {
+    case logType.EXTENSION_REQUESTS:
+      return formatExtensionRequestsLog(logs, usersMap);
+    case logType.REQUEST_CREATED:
+      return formatRequestCreatedLogs(logs, usersMap);
+    case logType.TASK:
+      return formatTaskUpdateLogs(logs, usersMap);
+    case logType.PROFILE_DIFF_APPROVED:
+      return formatProfileDiffLogs(logs, usersMap, logType.PROFILE_DIFF_APPROVED);
+    case logType.PROFILE_DIFF_REJECTED:
+      return formatProfileDiffLogs(logs, usersMap, logType.PROFILE_DIFF_REJECTED);
+    default:
+      return {};
+  }
+}
+
```

Comment:
repeated logic for formatting different types of logs. Could you consider refactoring this to reduce redundancy and improve readability?